File: src/main/java/no/nav/pensjon/innsyn/domain/Forstegangstjeneste.java
package no.nav.pensjon.innsyn.domain;

public class Forstegangstjeneste {

    private final String tjenestestart;
    private final String dimittert;
    private final String rapporttype;
    private final String status;
    private final String kilde;

    public Forstegangstjeneste(String tjenestestart, String dimittert, String rapporttype, String status, String kilde) {
        this.tjenestestart = tjenestestart;
        this.dimittert = dimittert;
        this.rapporttype = rapporttype;
        this.status = status;
        this.kilde = kilde;
    }

    public String getTjenestestart() {
        return tjenestestart;
    }

    public String getDimittert() {
        return dimittert;
    }

    public String getRapporttype() {
        return rapporttype;
    }

    public String getStatus() {
        return status;
    }

    public String getKilde() {
        return kilde;
    }
}

File: src/main/kotlin/no/nav/pensjon/innsyn/domain/popp/Forstegangstjeneste.kt
package no.nav.pensjon.innsyn.domain.popp

import no.nav.pensjon.innsyn.domain.Domain
import no.nav.pensjon.innsyn.domain.popp.support.FTjenStatus
import no.nav.pensjon.innsyn.domain.popp.support.KildeType
import no.nav.pensjon.innsyn.domain.popp.support.RapportType
import java.time.LocalDate
import javax.persistence.*

@Entity
@Table(name = "T_F_TJEN_TOT")
data class Forstegangstjeneste(
        @Id
        @Column(name = "PERSON_ID")
        private var personId: Int,
        @Column(name = "DATO_TJENESTESTART")
        val tjenestestart: LocalDate,
        @Column(name = "DATO_DIMITTERING")
        val dimittert: LocalDate,
        @ManyToOne
        @JoinColumn(name = "K_RAPPORT_T")
        private val rapportType: RapportType,
        @ManyToOne
        @JoinColumn(name = "K_F_TJEN_TOT_S")
        private val fTjenStatus: FTjenStatus,
        @ManyToOne
        @JoinColumn(name = "K_KILDE_T")
        private val kildeType: KildeType
): Domain {
    @Transient
    val rapporttype = rapportType.dekode
    @Transient
    val status = fTjenStatus.dekode
    @Transient
    val kilde = kildeType.dekode
    @Transient
    override val fields = setOf(::tjenestestart, ::dimittert, ::rapporttype, ::status, ::kilde)
}